docs/core/misc/slip0039.md

```diff
@@ -0,0 +1,105 @@
+# Use of SLIP-39 in trezor-core
+
+[SLIP-39](https://github.com/satoshilabs/slips/blob/master/slip-0039.md) describes a way
+to securely back up a secret value using Shamir's Secret Sharing scheme.
+
+The secret value, called a Master Secret (**MS**) in SLIP-39 terminology, is first
+encrypted by a passphrase, producing an Encrypted Master Secret (**EMS**). The EMS is
+then split into a number of shares, which are encoded as a set of mnemonic words.
+Afterwards, it is possible to recombine some or all of the shares to obtain back the
+EMS, and when the correct passphrase is provided, decrypt the original Master Secret.
+
+This does not quite match Trezor's use of the "passphrase protection" feature, namely
+that any passphrase is valid, and using any passphrase will yield a working wallet.
+
+SLIP-39 enables this usage by specifying that passphrases are not validated in any way.
+Decrypting an EMS with any passphrase will produce data usable as the Master Secret,
+regardless of whether it is the original data or not.
+
+
+## Seed handling in Trezor
+
+Trezor stores a _mnemonic secret_ in a storage field `_MNEMONIC_SECRET`. This is the
+input for the root node derivation process: `mnemonic.get_seed(passphrase)` takes the
+user-provided passphrase as an argument, and derives the appropriate root node from the
+mnemonic secret.
+
+With BIP-39, the recovery phrase itself is the mnemonic secret. During device
+initialization, the raw recovery phrase is given to the user, and also directly stored
+in the `_MNEMONIC_SECRET` field. Whenever the root node is required, it is derived by
+applying PBKDF2 to the mnemonic secret plus passphrase.
+
+For SLIP-39 it is not practical to store the raw data of the recovery shares. During
+device initialization, a random Encrypted Master Secret is generated and stored as
+`_MNEMONIC_SECRET`. SLIP-39 encryption parameters (a random identifier and an iteration
+exponent) are stored alongside the mnemonic secret in their own storage fields. Whenever
+the root node is required, it is derived by "decrypting" the stored mnemonic secret with
+the provided passphrase.
```

Comment:
I'm wondering if adding a rationale why we call it "encrypt"/"decrypt" might make things clearer. Something like: "In BIP-39 we use the verb `derive` to denote the process of applying a passphrase. In SLIP-39 this process is symmetric - both directions are possible. Therefore we were reluctant to keep the name "derive".